src/compose/service.coffee

```diff
@@ -0,0 +1,709 @@
+				OomScoreAdj: @oomScoreAdj
+				OomKillDisable: @oomKillDisable
+				Tmpfs: tmpfs
+				Dns: @dns
+				DnsSearch: @dnsSearch
+				DnsOpt: @dnsOpt
+				Ulimits: @ulimitsArray
+				ReadonlyRootfs: @readOnly
+				Sysctls: @sysctls
+				CgroupParent: @cgroupParent
+				ExtraHosts: @extraHosts
+				GroupAdd: @groupAdd
+				PidMode: @pid
+				PidsLimit: @pidsLimit
+				SecurityOpt: @securityOpt
+				UsernsMode: @usernsMode
+				IpcMode: @ipc
+		}
+		if @stopSignal?
+			conf.StopSignal = @stopSignal
+		if @stopGracePeriod?
+			conf.StopTimeout = @stopGracePeriod
+		if @healthcheck?
+			conf.Healthcheck = @healthcheck
+		if @restartPolicy.Name != 'no'
+			conf.HostConfig.RestartPolicy = @restartPolicy
+		# If network mode is the default network for this app, add alias for serviceName
+		if @networkMode == "#{@appId}_default"
+			conf.NetworkingConfig = {
+				EndpointsConfig: {
+					"#{@appId}_default": {
+						Aliases: [ @serviceName ]
+					}
+				}
+			}
+		if @init
+			conf.HostConfig.Init = true
+		if !_.isEmpty(@hostname)
+			conf.Hostname = @hostname
+		if !_.isEmpty(@storageOpt)
+			conf.HostConfig.StorageOpt = @storageOpt
+		if @macAddress?
+			conf.MacAddress = @macAddress
+		return conf
+
+	# TODO: when we support network configuration properly, return endpointConfig: conf
+	extraNetworksToJoin: =>
+		_.map _.pickBy(@networks, (conf, net) => net != @networkMode), (conf, net) ->
+			return { name: net, endpointConfig: {} }
+
+	# TODO: compare configuration, not only network names
+	hasSameNetworks: (otherService) =>
+		_.isEmpty(_.xor(_.keys(@networks), _.keys(otherService.networks)))
+
+	isSameContainer: (otherService) =>
+		propertiesToCompare = [
+			'image'
+			'command'
+			'entrypoint'
+			'networkMode'
+			'privileged'
+			'restartPolicy'
+			'labels'
+			'portBindings'
+			'exposedPorts'
+			#'memLimit'
+			#'memReservation'
+			'shmSize'
+			'cpuShares'
+			'cpuQuota'
+			'nanoCpus'
+			'cpuset'
+			'domainname'
+			'oomScoreAdj'
+			'oomKillDisable'
+			'healthcheck'
+			'stopSignal'
+			'stopGracePeriod'
+			'init'
+			'readOnly'
+			'sysctls'
+			'hostname'
+			'cgroupParent'
+			'pid'
+			'pidsLimit'
+			'storageOpt'
+			'usernsMode'
+			'ipc'
+			'macAddress'
+			'user'
+		]
+		arraysToCompare = [
+			'volumes'
+			'devices'
+			'capAdd'
+			'capDrop'
+			'dns'
+			'dnsSearch'
+			'dnsOpt'
+			'tmpfs'
+			'extraHosts'
+			'ulimitsArray'
+			'groupAdd'
+			'securityOpt'
+		]
+		isEq = _.isEqual(_.pick(this, propertiesToCompare), _.pick(otherService, propertiesToCompare)) and
+			_.isEqual(_.omit(@environment, [ 'RESIN_DEVICE_NAME_AT_INIT' ]), _.omit(otherService.environment, [ 'RESIN_DEVICE_NAME_AT_INIT' ])) and
+			@hasSameNetworks(otherService) and
+			_.every arraysToCompare, (property) =>
+				_.isEmpty(_.xorWith(this[property], otherService[property], _.isEqual))
+
+		# This can be very useful for debugging so I'm leaving it commented for now.
+		# Uncomment to see the services whenever they don't match.
+		#if !isEq
```

Comment:
This should be removed before merging or hidden behind a `DEBUG` flag (which can be set to be compiled out by webpack for release versions)